fib-prime.js

```diff
@@ -1,34 +1,38 @@
-var ispnum = function(num) {
-  for(var i = 2; i < num; i++)
-    if  (num % i === 0) return false;
-  return num > 1;
-};
-
-const fibonacci = (num) => {
-  if (num <= 1)return 1;
-  return fibonacci(num - 1) + fibonacci(num - 2);
-};
-
-function nxtPrmFib(number) {
-  let r = 0;
-  let l = 1;
+const isPrime = num => {
+  for (let i = 2, s = Math.sqrt(num); i <= s; i++)
+    if (num % i === 0) return false
+
+  return num > 1
+}
+
+const fibonacci = num => {
+  let phi = (1 + Math.sqrt(5)) / 2
+  return Math.round(Math.pow(phi, num) / Math.sqrt(5))
```

Comment:
Optimized Fibonacci function with `Fn = {[(√5 + 1)/2] ^ n} / √5 ` method,
Can be changed to:
`let f = new Array(num + 2)
  let i
  f[0] = 0
  f[1] = 1
  for (i = 2; i <= num; i++) {
    f[i] = f[i - 1] + f[i - 2]
  }
  return f[num]`